File: CS/Dashboard_UnderlyingDataWin/Form1.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardWin;
using DevExpress.Utils;
using DevExpress.XtraEditors;
using System.Windows.Forms;

namespace Dashboard_UnderlyingDataWin {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
        }

        private void dashboardViewer1_DashboardItemDoubleClick(object sender, DashboardItemMouseActionEventArgs e) {
            XtraForm form = new XtraForm {
                Text = "Underlying Data"
            };
            DashboardUnderlyingDataSet underlyingData = e.GetUnderlyingData();

            if (underlyingData != null && underlyingData.RowCount > 0) {
                DevExpress.XtraGrid.GridControl grid = new DevExpress.XtraGrid.GridControl {
                    Parent = form,
                    Dock = DockStyle.Fill,
                    DataSource = underlyingData,
                };
            }
            else {
                LabelControl lbl = new LabelControl {
                    Text = "No Data",
                    Parent = form,
                };
                lbl.AutoSizeMode = LabelAutoSizeMode.None;
                lbl.Appearance.TextOptions.HAlignment = HorzAlignment.Center;
                lbl.Appearance.TextOptions.VAlignment = VertAlignment.Center;
                lbl.Dock = DockStyle.Fill;
            }

            form.ShowDialog();
            form.Dispose();
        }
    }
}

File: VB/Dashboard_UnderlyingDataWin/Form1.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWin
Imports DevExpress.Utils
Imports DevExpress.XtraEditors
Imports System.Windows.Forms

Namespace Dashboard_UnderlyingDataWin
	Partial Public Class Form1
		Inherits XtraForm

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub dashboardViewer1_DashboardItemDoubleClick(ByVal sender As Object, ByVal e As DashboardItemMouseActionEventArgs) Handles dashboardViewer1.DashboardItemDoubleClick
			Dim form As XtraForm = New XtraForm With {.Text = "Underlying Data"}
			Dim underlyingData As DashboardUnderlyingDataSet = e.GetUnderlyingData()


			If underlyingData IsNot Nothing AndAlso underlyingData.RowCount > 0 Then
				Dim grid As DevExpress.XtraGrid.GridControl = New DevExpress.XtraGrid.GridControl With {.Parent = form, .Dock = DockStyle.Fill, .DataSource = underlyingData}
			Else
				Dim lbl As LabelControl = New LabelControl With {.Text = "No Data", .Parent = form}
				lbl.AutoSizeMode = LabelAutoSizeMode.None
				lbl.Appearance.TextOptions.HAlignment = HorzAlignment.Center
				lbl.Appearance.TextOptions.VAlignment = VertAlignment.Center
				lbl.Dock = DockStyle.Fill
			End If

			form.ShowDialog()
			form.Dispose()
		End Sub
	End Class
End Namespace
